src/RequestInsurance/ViewComponents/PrettyPrintDifference.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+namespace Cego\RequestInsurance\ViewComponents;
+
+use Exception;
+use Illuminate\View\View;
+use Jfcherng\Diff\Differ;
+use Jfcherng\Diff\DiffHelper;
+use Illuminate\View\Component;
+use Jfcherng\Diff\Factory\RendererFactory;
+
+class PrettyPrintDifference extends Component
+{
+    /**
+     * String with HTML displaying the difference between old string and new string.
+     *
+     * @var string
+     */
+    public $content;
+
+    /**
+     * Rendering options used when creating string of HTML.
+     *
+     * @var array
+     */
+    protected array $rendererOptions = [
+        'detailLevel'         => 'word',
+        'showHeader'          => true,
+        'separateBlock'       => true,
+        'resultForIdenticals' => null,
+        'lineNumbers'         => false,
+    ];
+
+    /**
+     * Additional options used when looking for differences in the two given strings.
+     *
+     * @var array
+     */
+    protected array $differOptions = [
+        'ignoreWhitespace' => false,
+        'context'          => 3,
+    ];
+
+    /**
+     * Create a new component instance.
+     *
+     * @param object|array|string $content
+     *
+     * @throws \JsonException
+     */
+    public function __construct($oldValues, $newValues)
+    {
+        $this->content = $this->prettyPrint($oldValues, $newValues);
+    }
+
+    /**
+     * Get the view / contents that represent the component.
+     *
+     * @return \Illuminate\Contracts\View\Factory|View
+     */
+    public function render()
+    {
+        return view('request-insurance::components.pretty-print-difference');
+    }
+
+    /**
+     * Pretty prints the difference between an old string and a new string.
+     *
+     * @param string $oldContent
+     * @param string $newContent
+     *
+     * @return string
+     */
+    protected function prettyPrint(string $oldContent, string $newContent): string
+    {
+        try {
+            // We need to use different renderering options for capturing differences in json, otherwise the result is quite useless
+            if ($this->validJson($oldContent)) {
+                return $this->prettyPrintDifferenceJson($oldContent, $newContent);
+            }
+
+            // DiffHelper returns a string in html format.
+            $content = DiffHelper::calculate($oldContent, $newContent, 'Json', $this->differOptions);
```

Comment:
Hvad sker der, hvis man ændre i noget andet end json? Altså f.eks. XML?

Reply:
Jeg har testet det med noget simpelt XML, og det virkede fint. Og det virker også fint, hvis det bare er simple strings. Json her refererer til output formatet for calculate, så min kommentar var noget misvisende 😄